File: UsingParceler/app/src/main/java/github/nisrulz/sample/usingparceler/DetailsActivity.java
package github.nisrulz.sample.usingparceler;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.Toast;
import org.parceler.Parcels;

public class DetailsActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_details);

    // Use utility function Parcels#unwrap() to convert from parceable to java bean
    Person person = Parcels.unwrap(
        getIntent().getExtras().getParcelable(getString(R.string.person_detail_key)));

    Toast.makeText(this, person.toString(), Toast.LENGTH_SHORT).show();
  }
}

File: UsingParcelize/app/src/main/java/github/nisrulz/usingparcelize/DetailsActivity.kt
package github.nisrulz.usingparcelize

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Parcelable
import android.widget.Toast

class DetailsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_details)

        val person = intent.extras?.getParcelable<Parcelable>(getString(R.string.person_detail_key)) as Person?
        person?.let {
            Toast.makeText(this, person.toString(), Toast.LENGTH_SHORT).show();
        }
    }
}
